academic-services/knowledge-exploration-service/reference-makes-command-line-tool.md

```diff
@@ -171,37 +197,288 @@ The input entities file Url prefix.
+
+`--SchemaFilePath`
+
+The input schema json file path. E.g. indexSchema.json
+
+`--EntitiesFilePath`
+
+The input entities json file path. E.g. indexEntities.json
+
+`--OutputIndexFilePath`
+
+The output MAKES index binary file path. E.g. index.kes
+
+### BuildIndexLocal optional parameters
+
+`--IndexDescription`
+
+A description to include in the index. E.g. My custom MAKES index build.
+
+`--MaxStringLength`
+
+The maximum string length for all entity attributes. All strings over the maximum string length will be truncated
+
+`--IntersectionCountThresholdForPreCompute`
+
+The attribute value intersections threshold for pre-computing look up tables. Use this value to tune index build time performance, run time performance, and index size. The higher the value, the smaller index size and slower run-time performance will be. The lower the value, the larger index size and faster run-time performance will be.
+
+## CompileGrammarLocal command
+
+Compiles a grammar definition xml file into compiled grammar file.
+
+```cmd
+kesm CompileGrammarLocal --GrammarDefinitionFilePath
+                         --OutputCompiledGrammarFilePath
+```
+
+### CompileGrammarLocal required parameters
+
+`--GrammarDefinitionFilePath`
+
+The input grammar definition xml file path. E.g. grammar.xml
+
+`--OutputCompiledGrammarFilePath`
+
+The output compiled grammar binary file path. E.g. grammar.kesg
+
+## DescribeIndex command
+
+Retrieves the description, schema, build time and number of entities for the index binary.
+
+```cmd
+kesm DescribeIndex --IndexFilePath
+```
+
+### DescribeIndex required parameters
+
+`--IndexFilePath`
+
+The file path to the index binary. E.g. index.kes
+
+## DescribeGrammar command
+
+Retrieves original grammar definition XML from the compiled grammar binary
+
+```cmd
+kesm DescribeGrammar --CompiledGrammarFilePath
+```
+
+### DescribeGrammar required parameters
+
+`--CompiledGrammarFilePath`
+
+The file path to the compiled grammar binary. E.g. grammar.kesg
+
+## Interpret command
+
+Interprets a natural language query using specified indexes and grammar.
+
+```cmd
+kesm Interpret --Query
+               --IndexFilePaths
+               --GrammarFilePath
+               [--NormalizeQuery]
+               [--AllowCompletions]
+               [--Offset]
+               [--Count]
+               [--InterpretationEntityAttributes]
+               [--InterpretationEntityCount]
+               [--Timeout]
+```
+
+### Interpret required parameters
+
+`--Query`
+
+The natural lanugage query string to interpret.
+
+`--IndexFilePaths`
+
+The file path expression for specifying which index file(s) to use. Use wild card to specify multiple indexes e.g. './index.*.kes'.
+
+`--GrammarFilePath`
+
+The compiled grammar binary file path. Interpret requires a compiled grammar binary
+
+### Interpret optional parameters
+
+`--NormalizeQuery`
+
+Whether normalization rules should be applied to the query before making interpretations.
```

Comment:
Do we describe what normalization rules are somewhere?  Can we add a pointer?

Reply:
[https://docs.microsoft.com/en-us/academic-services/knowledge-exploration-service/concepts-queries#stage-1-lexical-analysis](https://docs.microsoft.com/en-us/academic-services/knowledge-exploration-service/concepts-queries#stage-1-lexical-analysis)

---
In reply to: [465191614](https://github.com/MicrosoftDocs/microsoft-academic-services/pull/172#discussion_r465191614) [](ancestors = 465191614)